lxd/container_lxc.go

```diff
@@ -3508,14 +3522,18 @@ func (c *containerLXC) getDiskLimits() (map[string]deviceBlockLimit, error) {
 		device := deviceBlockLimit{readBps: readBps, readIops: readIops, writeBps: writeBps, writeIops: writeIops}
 
 		for _, block := range blocks {
-			dev := strings.TrimPrefix(block, "/dev/")
+			if !strings.HasPrefix(block, "/dev/") {
+				continue
+			}
 
+			dev := strings.TrimPrefix(block, "/dev/")
 			if strings.Contains(dev, "/") {
 				continue
 			}
 
 			if !shared.PathExists(fmt.Sprintf("/sys/class/block/%s/dev", dev)) {
-				return nil, fmt.Errorf("Disk %s is missing /sys/class/block entry", dev)
+				shared.Log.Error("Disk is missing /sys/class/block entry", log.Ctx{"device": dev})
+				continue
```

Comment:
So in theory we could only keep the hasLimits part of this change and drop all other checks, which would then make things fail if setting a limit on a device which doesn't support it.

I'll try and rework this mess when I'm a bit more awake (just got off a plane).